app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt

```diff
@@ -0,0 +1,28 @@
+package com.wire.android.notification
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.wire.android.appLogger
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.kalium.logic.CoreLogic
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class MyFirebaseMessagingService : FirebaseMessagingService() {
+
+    @Inject
+    @KaliumCoreLogic
+    lateinit var coreLogic: CoreLogic
+
+    override fun onNewToken(p0: String) {
+        super.onNewToken(p0)
+        coreLogic.getAuthenticationScope().saveNotificationToken(p0, "GCM")
```

Comment:
What if it returns failure? Maybe we could use `WorkManager` to schedule this request when there is internet access for instance

Reply:
actually I was thinking to re try to generate the token in the login again , in case this one fail to save it when we first launch the app , but the work manager sounds a good idea as well , I can experiment with that